Mathlib/Algebra/Group/Units.lean

```diff
@@ -31,29 +35,705 @@ universe u
+
+/-- `field_simp` needs the reverse direction of `divp_assoc` to move all `/ₚ` to the right. -/
+@[field_simps]
+theorem divp_assoc' (x y : α) (u : αˣ) : x * (y /ₚ u) = x * y /ₚ u :=
+  (divp_assoc _ _ _).symm
+
+@[simp]
+theorem divp_inv (u : αˣ) : a /ₚ u⁻¹ = a * u :=
+  rfl
+
+@[simp]
+theorem divp_mul_cancel (a : α) (u : αˣ) : a /ₚ u * u = a :=
+  (mul_assoc _ _ _).trans <| by rw [Units.inv_mul, mul_one]
+
+@[simp]
+theorem mul_divp_cancel (a : α) (u : αˣ) : a * u /ₚ u = a :=
+  (mul_assoc _ _ _).trans <| by rw [Units.mul_inv, mul_one]
+
+@[simp]
+theorem divp_left_inj (u : αˣ) {a b : α} : a /ₚ u = b /ₚ u ↔ a = b :=
+  Units.mul_left_inj _
+
+@[field_simps]
+theorem divp_divp_eq_divp_mul (x : α) (u₁ u₂ : αˣ) : x /ₚ u₁ /ₚ u₂ = x /ₚ (u₂ * u₁) := by
+  simp only [divp, mul_inv_rev, Units.val_mul, mul_assoc]
+
+@[field_simps]
+theorem divp_eq_iff_mul_eq {x : α} {u : αˣ} {y : α} : x /ₚ u = y ↔ y * u = x :=
+  u.mul_left_inj.symm.trans <| by rw [divp_mul_cancel]; exact ⟨Eq.symm, Eq.symm⟩
+
+@[field_simps]
+theorem eq_divp_iff_mul_eq {x : α} {u : αˣ} {y : α} : x = y /ₚ u ↔ x * u = y := by
+  rw [eq_comm, divp_eq_iff_mul_eq]
+
+theorem divp_eq_one_iff_eq {a : α} {u : αˣ} : a /ₚ u = 1 ↔ a = u :=
+  (Units.mul_left_inj u).symm.trans <| by rw [divp_mul_cancel, one_mul]
+
+@[simp]
+theorem one_divp (u : αˣ) : 1 /ₚ u = ↑u⁻¹ :=
+  one_mul _
+
+/-- Used for `field_simp` to deal with inverses of units. -/
+@[field_simps]
+theorem inv_eq_one_divp (u : αˣ) : ↑u⁻¹ = 1 /ₚ u := by rw [one_divp]
+
+/-- Used for `field_simp` to deal with inverses of units. This form of the lemma
+is essential since `field_simp` likes to use `inv_eq_one_div` to rewrite
+`↑u⁻¹ = ↑(1 / u)`.
+-/
+@[field_simps]
+theorem inv_eq_one_divp' (u : αˣ) : ((1 / u : αˣ) : α) = 1 /ₚ u := by
+  rw [one_div, one_divp]
+
+/-- `field_simp` moves division inside `αˣ` to the right, and this lemma
+lifts the calculation to `α`.
+-/
+@[field_simps]
+theorem val_div_eq_divp (u₁ u₂ : αˣ) : ↑(u₁ / u₂) = ↑u₁ /ₚ u₂ := by
+  rw [divp, division_def, Units.val_mul]
+#align coe_div_eq_divp val_div_eq_divp
+
+end Monoid
+
+section CommMonoid
+
+variable [CommMonoid α]
+
+@[field_simps]
+theorem divp_mul_eq_mul_divp (x y : α) (u : αˣ) : x /ₚ u * y = x * y /ₚ u := by
+  rw [divp, divp, mul_right_comm]
+
+-- Theoretically redundant as `field_simp` lemma.
+@[field_simps]
+theorem divp_eq_divp_iff {x y : α} {ux uy : αˣ} : x /ₚ ux = y /ₚ uy ↔ x * uy = y * ux := by
+  rw [divp_eq_iff_mul_eq, divp_mul_eq_mul_divp, divp_eq_iff_mul_eq]
+
+-- Theoretically redundant as `field_simp` lemma.
+@[field_simps]
+theorem divp_mul_divp (x y : α) (ux uy : αˣ) : x /ₚ ux * (y /ₚ uy) = x * y /ₚ (ux * uy) := by
+  rw [divp_mul_eq_mul_divp, divp_assoc', divp_divp_eq_divp_mul]
+
+end CommMonoid
+
+/-!
+# `IsUnit` predicate
+
+In this file we define the `IsUnit` predicate on a `Monoid`, and
+prove a few basic properties. For the bundled version see `Units`. See
+also `Prime`, `Associated`, and `Irreducible` in `Algebra.Associated`.
```

Comment:
This seems way out of date; it's all one file now.

Perhaps change to `Mathlib.Algebra.Associated`.